src/routing/mod.rs

```diff
@@ -213,13 +254,54 @@ impl<S> Router<S> {
     /// making the type easier to name. This is sometimes useful when working with
     /// `nest`.
     ///
+    /// # Wildcard routes
+    ///
+    /// Nested routes are similar to wildcard routes. The difference is that
+    /// wildcard routes still see the whole URI whereas nested routes will have
+    /// the prefix stripped.
+    ///
+    /// ```rust
+    /// use axum::{handler::get, http::Uri, Router};
+    ///
+    /// let app = Router::new()
+    ///     .route("/foo/*rest", get(|uri: Uri| async {
+    ///         // `uri` will contain `/foo`
+    ///     }))
+    ///     .nest("/bar", get(|uri: Uri| async {
+    ///         // `uri` will _not_ contain `/bar`
+    ///     }));
+    /// # async {
+    /// # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
+    /// # };
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// Panics if the route overlaps with another route. See [`Router::route`]
+    /// for more details.
+    ///
     /// [`OriginalUri`]: crate::extract::OriginalUri
-    pub fn nest<T>(self, path: &str, svc: T) -> Router<Nested<T, S>> {
-        self.map(|fallback| Nested {
-            pattern: PathPattern::new(path),
-            svc,
-            fallback,
-        })
+    pub fn nest<T>(mut self, path: &str, svc: T) -> Router<Nested<T, S>> {
+        let id = RouteId::next();
+
+        let path = if path == "/" {
+            format!("/*{}", NEST_TAIL_PARAM)
+        } else {
+            format!("{}/*{}", path, NEST_TAIL_PARAM)
```

Comment:
Would nesting with "/{}" instead of "/*{}" also work?
Then nesting would behave more like other routes and users still could Wildcard routes themself.

Reply:
No. Nested routes must match all routes start with the prefix given by the user. That is what this does.

Reply:
Ok thanks